spec/support/email_helpers.rb

```diff
@@ -0,0 +1,21 @@
+module EmailHelpers
+  def stub_welcome_email_delivery
+    welcome_email_delivery = instance_double(ActionMailer::MessageDelivery)
+    allow(UserMailer).to receive(:welcome)
+      .and_return(welcome_email_delivery)
+    allow(welcome_email_delivery).to receive(:deliver_later)
+  end
+
+  def expect_welcome_email_delivery(user: nil)
+    welcome_email_delivery = instance_double(ActionMailer::MessageDelivery)
+    if user
+      expect(UserMailer).to receive(:welcome)
+        .with(user.id)
+        .and_return(welcome_email_delivery)
+    else
```

Comment:
When would we expect a welcome email without an attached user?

Reply:
In our feature test where I want to expect a welcome email is sent but there is no `user` record set up to use in the expectation, since the submit button has not been clicked at that point.

Can you think of how we might intercept the User in a concise way? Stubbing the User creation seemed too convulted and messy. I do agree that it would be a better test if we could asset that an email was sent to a given email address.

Perhaps we could send the email address in as a parameter solely for the purposes of testing?
```
expect(UserMailer).to receive(:welcome)
  .with(user.id, foo@bar.com)

UserMailer
  def welcome(id, email)
    user = User.find id
    email = email # This would be better to use the user.email to avoid any weird state problem
```